#! /bin/sh /usr/share/dpatch/dpatch-run
## opt-45_yaepg.dpatch by bball950@yahoo.com
## http://www.hoochvdr.info/yaepg/vdr-1.3.19-core-yaepg.diff
##
## Thomas Günther <tom@toms-cafe.de>:
##   - adapted to VDR-1.3.25
##   - adapted to VDR-1.3.42
##   - adapted to VDR-1.5.2
##   - adapted to VDR-1.5.3 (removed special yaepg font)
##   - adapted to VDR-1.5.9
##   - adapted to VDR-1.7.11
## Tobias Grimm <tg@e-tobi.net>:
##   - applied modification from Michaël Nival that fixes a small bug
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch is needed for the yaepg plugin.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr~/PLUGINS/src/dvbsddevice/dvbsdffosd.c vdr/PLUGINS/src/dvbsddevice/dvbsdffosd.c
--- vdr~/PLUGINS/src/dvbsddevice/dvbsdffosd.c	2011-01-27 23:35:24.000000000 +0100
+++ vdr/PLUGINS/src/dvbsddevice/dvbsdffosd.c	2011-03-19 22:55:53.000000000 +0100
@@ -78,6 +78,10 @@
             Cmd(OSD_SetWindow, 0, i + 1);
             Cmd(OSD_Close);
             }
+        if (vidWin.bpp != 0) {
+           Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+           Cmd(OSD_Close);
+           }
         shown = false;
         }
      }
@@ -113,6 +117,10 @@
          Cmd(OSD_SetWindow, 0, i + 1);
          Cmd(OSD_Close);
          }
+     if (vidWin.bpp != 0) {
+        Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+        Cmd(OSD_Close);
+        }
      shown = false;
      }
   return cOsd::SetAreas(Areas, NumAreas);
@@ -196,6 +204,11 @@
          Cmd(OSD_SetWindow, 0, i + 1);
          Cmd(OSD_MoveWindow, 0, Left() + Bitmap->X0(), Top() + Bitmap->Y0());
          }
+     if (vidWin.bpp != 0) {
+        Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+        Cmd(OSD_OpenRaw, vidWin.bpp, vidWin.x1, vidWin.y1,
+            vidWin.x2, vidWin.y2, (void *)0);
+        }
      shown = true;
      }
 }
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr~/device.h vdr/device.h
--- vdr~/device.h	2011-03-19 22:55:53.000000000 +0100
+++ vdr/device.h	2011-03-19 22:55:53.000000000 +0100
@@ -271,12 +271,12 @@
          ///< Direction (only the sign of Direction is evaluated, positive values
          ///< switch to higher channel numbers).
 private:
-  eSetChannelResult SetChannel(const cChannel *Channel, bool LiveView);
-         ///< Sets the device to the given channel (general setup).
 protected:
   virtual bool SetChannelDevice(const cChannel *Channel, bool LiveView);
          ///< Sets the device to the given channel (actual physical setup).
 public:
+  eSetChannelResult SetChannel(const cChannel *Channel, bool LiveView);
+         ///< Sets the device to the given channel (general setup).
   static int CurrentChannel(void) { return primaryDevice ? currentChannel : 0; }
          ///< Returns the number of the current channel on the primary device.
   static void SetCurrentChannel(const cChannel *Channel) { currentChannel = Channel ? Channel->Number() : 0; }
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr~/osd.c vdr/osd.c
--- vdr~/osd.c	2011-03-19 21:07:09.000000000 +0100
+++ vdr/osd.c	2011-03-19 22:55:53.000000000 +0100
@@ -1596,6 +1596,7 @@
          }
       }
   Osds.Append(this);
+  vidWin.bpp = 0;
 }
 
 cOsd::~cOsd()
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr~/osd.h vdr/osd.h
--- vdr~/osd.h	2011-03-19 21:07:09.000000000 +0100
+++ vdr/osd.h	2011-03-19 22:56:45.000000000 +0100
@@ -917,6 +917,7 @@
        ///<        MyOsdDrawPixmap(Left() + pm->ViewPort().X(), Top() + pm->ViewPort().Y(), pm->Data(), w, h, h * d);
        ///<        delete pm;
        ///<        }
+  tArea vidWin;
   };
 
 #define MAXOSDIMAGES 64
